src/coredump/coredump-ratelimit.c

```diff
@@ -0,0 +1,149 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <sys/statvfs.h>
+
+#include "sd-id128.h"
+
+#include "coredump-ratelimit.h"
+#include "dirent-util.h"
+#include "fd-util.h"
+#include "fs-util.h"
+#include "missing_sched.h"
+#include "string-util.h"
+#include "time-util.h"
+#include "user-util.h"
+#include "util.h"
+#include "xattr-util.h"
+
+static int exe_from_file_name(const char *path, const char *file_name, char **ret) {
+        int r = 0;
+        _cleanup_free_ char *k = NULL;
+
+        assert(path);
+        assert(file_name);
+
+        k = path_join(path, file_name);
```

Comment:
Missing OOM check. Even better, just use `k = prefix_roota(path, filename);` so that it's stack-allocated instead (and remove `_cleanup_free_ from `k`'s declaration). Same below.

Reply:
Thanks for the comment. Will address the comment

Reply:
Currently busy with high-priority tasks and other issues.  Will check and update the review in a couple of weeks. 
Thank you for all the comments.